setup.py

```diff
@@ -0,0 +1,83 @@
+# BSD 3-Clause License
+#
+# Copyright (c) 2017, Science and Technology Facilities Council
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# Author R. Ford STFC Daresbury Lab.
+#
+"""Setup script. Used by easy_install and pip."""
+
+from setuptools import setup, find_packages
+
+PACKAGES = find_packages(where="src")
+
+NAME = 'melody'
+AUTHOR = 'Rupert Ford'
+AUTHOR_EMAIL = 'rupert.ford@stfc.ac.uk'
+URL = 'https://github.com/rupertford/melody'
+DOWNLOAD_URL = 'https://github.com/rupertford/melody'
+DESCRIPTION = 'melody'
+LONG_DESCRIPTION = '''melody: lightweight python parameter search software for performance analysis and tuning'''
```

Comment:
If you set your long description to something like:

open('README.rst').read()

where you replace "README.rst" with yours then you will be able to show the same page on pypi as you have on github including the badges. Unless of course you want pypi to have different information than the github one. Just a suggestion of what you can do here.